lib/chef/provider/execute.rb

```diff
@@ -69,6 +69,14 @@ def sensitive?
         !!new_resource.sensitive
       end
 
+      def stream_to_formatter?
+        Chef::Config[:always_stream_execute] || run_context.events.formatter?
```

Comment:
That makes sense, though I could see wanting to have a per-resource setting, some executes are more interesting than others.